### ISSUE TITLE
Move `ctx.session` to `ctx.state.session` to correctly type ctx.

### ISSUE THREAD

**Issue by c0per** [jcs224/oak_sessions#1]
When I use `app.use(session.initMiddleware());`, I'm having this error:
<img width="891" alt="image" src="https://user-images.githubusercontent.com/60544162/132270271-7bd7e1c5-2f64-4230-b132-fa6a8ec10ad1.png">

It looks like that SessionContext type of ctx in `session.initMiddleware()` seems to be the problem.

Plus, I need to add property `session` to type `Context` to avoid some ts errors.

I think move `ctx.session` to `ctx.state.session` wil solve the problem just fine.

**Comment by c0per** [comment]
or maybe add `ctx.state.session` as a refrence to `ctx.state` to implement the `Context` type in oak.

**Comment by jcs224** [comment]
Weird, works for me. What version of Oak are you using? Do you have an example I can see?

**Comment by jcs224** [comment]
Nevermind, I see now after converting my server test script from JS to TS. I'll look into it.

**Comment by c0per** [comment]
It's a TS type error. The code still functions well.

**Comment by c0per** [comment]
<img width="777" alt="image" src="https://user-images.githubusercontent.com/60544162/132275818-7bc90189-240a-4364-a835-76e76a43489a.png">

Type `SessionContext` isn't assignable to `Context` (oak requires ctx to be `Context` in their code) due to the extra `session` property.

I think the existing `state` property in `Context` is a good place to store session of a ctx.

It's better to just reference `ctx.session` in `ctx.state.session` to avoid a breaking change.

**Comment by c0per** [comment]
I made a PR to do the small change. #2

**Comment by jcs224** [comment]
Thanks. I think it's a good idea to use the `.state` property, it works properly with TS now.

However, it doesn't seem this will fix the user-facing code when actually implemented as per #2.

In my local testing, I need export the SessionContext interface, right now it's not available to any other file:

```ts
export interface SessionContext extends Context
```

Import the SessionContext
```ts
import { SessionContext } from './src/Session.ts' 
```
Then, the user has to attach the new context type to their router context somehow:
```ts
.get("/", session.initMiddleware(), async (context : SessionContext) => {})
```
So I think I'm just going to cut the cord on backward compatibility and do a new breaking version to 3.0. The library is still young, this is a change I want to make early.

Thanks!

**Comment by jcs224** [comment]
For now, I've included your hotfix, and I updated the readme, so it should allow TS users to be able to use the library without breaking JS user's stuff. However, for the next breaking version, I'm going to remove any references to `ctx.session`.
